src/sage/categories/tensor.py

```diff
@@ -51,7 +52,24 @@ class TensorProductFunctor(CovariantFunctorialConstruction):
     _functor_category = "TensorProducts"
     symbol = " # "
     unicode_symbol = f" {unicode_otimes} "
+    def __init__(self, category=None):
+        r"""
+        Constructor. See :class:`TensorProductFunctor` for details.
 
+        TESTS::
+
+            sage: from sage.categories.tensor import TensorProductFunctor
+            sage: TensorProductFunctor()
+            The tensor functorial construction
+        """
+        CovariantFunctorialConstruction.__init__(self)
+        self._forced_category = category
+        from sage.categories.rings import Rings
+        if self._forced_category is not None:
+            codomain = self._forced_category
+        else:
+            codomain = Rings()
+        MultivariateConstructionFunctor.__init__(self, Rings(), codomain)
```

Comment:
I added this blindly, by copy-pasting from `CartesianProductFunctor`.  `Rings()` is almost certainly wrong, I have no idea how the correct category is chosen.